webapp/src/App/Pages/Settings/Tokens/index.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState, useEffect } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import {isUserAdmin} from '../../../../services/utils'
+import { AppState } from '../../../reducers';
+import GenericSettingView from '../../../../components/Settings/GenericSettingView';
+import ItemSettings from '../../../../components/Settings/ItemSettings';
+import { getTokens, getTokensWithOwner, deleteToken} from '../../../../services/api/user';
+import {TokenDTO, IDType } from '../../../../common/types';
+import TokenModal from './TokenModal'
+import ConfirmationModal from '../../../../components/ui/ConfirmationModal'
+
+const mapState = (state: AppState) => ({
+    user: state.auth.user,
+  });
+
+const connector = connect(mapState);
+type ReduxProps = ConnectedProps<typeof connector>;
+type PropTypes = ReduxProps;
+
+const TokensSettings = (props: PropTypes) => {
+
+    const [t] = useTranslation();
+    const { enqueueSnackbar } = useSnackbar();
+    const [tokenList, setTokenList] = useState<Array<TokenDTO>>([]);
+    const [idForDeletion, setIdForDeletion] = useState<IDType>(-1);
+    const [filter, setFilter] = useState<string>("");
+    const {user} = props;
+
+    // Token modal
+    const [openModal, setOpenModal] = useState<boolean>(false);
+    const [openConfirmationModal, setOpenConfirmationModal] = useState<boolean>(false);
+    const [currentToken, setCurrentToken] = useState<TokenDTO|undefined>();
+
+    const createNewToken = () => {
+          setCurrentToken(undefined);
+          setOpenModal(true);
+    }
+
+    const onUpdateClick = (id: IDType) : void => {
+      setCurrentToken(tokenList.find((item)=> item.id === id));
+      setOpenModal(true);
+  }
+
+  const onDeleteClick = (id: IDType) => {
+    setIdForDeletion(id);
+    setOpenConfirmationModal(true);
+  }
+
+  const manageTokenDeletion = async () => {
+      try
+      {
+        const res = await deleteToken(idForDeletion as number);
+        setTokenList(tokenList.filter((item) => item.id !== idForDeletion));
+        console.log(res);
+        enqueueSnackbar(t('settings:onTokenDeleteSuccess'), { variant: 'success' });
+      }
+      catch(e)
+      {
+        enqueueSnackbar(t('settings:onTokenDeleteError'), { variant: 'error' });
+      }
+      setIdForDeletion(-1);
+      setOpenConfirmationModal(false);
+  }
+
+    const onModalClose = () => {
+      setOpenModal(false);
+    }
+
+    const onNewTokenCreation = (newToken : TokenDTO) : void => {
+      setTokenList(tokenList.concat(newToken));
+      setCurrentToken(newToken);
```

Comment:
here the token creation response will contain the jwt token.
it should be added to the token view with a subtext (a message in gray under it explaining the user that he should write down the token value since it won't reappear ever again).